test/src/test/java/org/corfudb/integration/AbstractIT.java

```diff
@@ -166,6 +170,28 @@ public void restartServer(CorfuRuntime corfuRuntime, String endpoint) {
         }
     }
 
+    /**
+     * Refreshes the layout and waits for a limited time for the refreshed layout to
+     * satisfy the expected verifier.
+     *
+     * @param verifier     Layout predicate to test the refreshed layout.
+     * @param corfuRuntime corfu runtime.
+     */
+    public static void waitForLayoutChange(Predicate<Layout> verifier, CorfuRuntime corfuRuntime) {
+        corfuRuntime.invalidateLayout();
+        Layout refreshedLayout = corfuRuntime.getLayoutView().getLayout();
+
+        for (int i = 0; i < PARAMETERS.NUM_ITERATIONS_MODERATE; i++) {
```

Comment:
I do not like these sleeps. Why do we not use latches or something more deterministic.

Reply:
If it is stuck the test will timeout and we will know something is wrong.